### PR TITLE
remove unused programs

### PR DESCRIPTION
**Pull request by dachen0**

### CODE REVIEW

File: docs/src/token-lending.md [hunk 1]
```diff
@@ -1,14 +0,0 @@
----
-title: Token-Lending Program
```
Comment (nope-finance) [inline docs/src/token-lending.md:2]: maybe dont delete this

File: token-lending/cli/Cargo.toml [hunk 2]
```diff
@@ -9,15 +9,15 @@ repository = "https://github.com/solana-labs/solana-program-library"
...
+clap = "=2.34.0"
+solana-clap-utils = "=1.7.12"
+solana-cli-config = "=1.7.12"
+solana-client = "=1.7.12"
+solana-logger = "=1.7.12"
+solana-sdk = "=1.7.12"
+solana-program = "=1.7.12"
 spl-token-lending = { path="../program", features = [ "no-entrypoint" ] }
-spl-token = { path="../../token/program", features = [ "no-entrypoint" ]  }
+spl-token = "=3.2.0"
```
Comment (nope-finance) [inline token-lending/cli/Cargo.toml]: ?

File: token-lending/cli/src/main.rs [hunk 3]
```diff
@@ -1168,8 +1168,8 @@ fn command_update_reserve(
     if pyth_price_pubkey.is_some() {
         println!(
             "Updating pyth oracle pubkey from {} to {}",
-            reserve.liquidity.pyth_oracle_pubkey.to_string(),
-            pyth_price_pubkey.unwrap().to_string(),
+            reserve.liquidity.pyth_oracle_pubkey,
```
Comment (nope-finance) [inline token-lending/cli/src/main.rs:1171]: why though?

File: token-lending/program/Cargo.toml [hunk 4]
```diff
@@ -16,19 +16,19 @@ arrayref = "0.3.6"
 bytemuck = "1.5.1"
 num-derive = "0.3"
 num-traits = "0.2"
-solana-program = "1.7.12"
-spl-token = { path = "../../token/program", features = [ "no-entrypoint" ] }
-switchboard-program = "0.1.45"
+solana-program = "=1.7.12"
+spl-token = { version = "3.1.0", features=["no-entrypoint"] }
```
Comment (nope-finance) [inline token-lending/program/Cargo.toml]: 3.3.0 ?

File: token-lending/cli/Cargo.toml (hunk 2 above)
Comment (dachen0) [reply token-lending/cli/Cargo.toml]: deleted the token program from the repo
pinned spl-token to v3.2.0

File: token-lending/cli/src/main.rs (hunk 3 above)
Comment (dachen0) [reply token-lending/cli/src/main.rs:1171]: clippy didn't like it
smth smth `This crap implements Display, you don't need to_string()`

File: token-lending/program/Cargo.toml (hunk 4 above)
Comment (dachen0) [reply token-lending/program/Cargo.toml]: um yea sure can do ig
was messing around with the spl-token version to figure out what was the breaking change

File: token-lending/cli/Cargo.toml (hunk 2 above)
Comment (nope-finance) [reply token-lending/cli/Cargo.toml]: y not 3.3 i mean and why remove feature

File: token-lending/program/tests/set_lending_market_owner.rs [hunk 5]
```diff
@@ -26,7 +26,7 @@ async fn test_success() {
     );
 
     // limit to track compute unit increase
-    test.set_bpf_compute_max_units(3_000);
+    test.set_bpf_compute_max_units(4_000);
```
Comment (DaSichuan) [inline token-lending/program/tests/set_lending_market_owner.rs]: lol why did this need to change?

Comment (DaSichuan) [inline token-lending/program/tests/set_lending_market_owner.rs]: lol why did this need to change?

**Comment by nope-finance** [comment]
solana version update slightly different compute calculations i guess.

**Comment by 0xkiplet** [comment]
This seems fine to me. Removing unused programs and pinning dependencies. I guess the `clap` dep was updated, but I think this is only used for the `cli` stuff.

**Pull request merged**